file_formats/cmt_format.ml

```diff
@@ -103,6 +144,239 @@ let clear_env binary_annots =
+        f ~namespace:Constructor env (Pident id) lid
+    | { Types.cstr_res; cstr_name; _ } ->
+        let path = path_in_type cstr_res cstr_name in
+        Option.iter (fun path -> f ~namespace:Constructor env path lid) path
+  in
+  let add_label env lid { Types.lbl_name; lbl_res; _ } =
+    let path = path_in_type lbl_res lbl_name in
+    Option.iter (fun path -> f ~namespace:Label env path lid) path
+  in
+  let with_constraint ~env (_path, _lid, with_constraint) =
+    match with_constraint with
+    | Twith_module (path', lid') | Twith_modsubst (path', lid') ->
+        f ~namespace:Module env path' lid'
+    | _ -> ()
+  in
+  Tast_iterator.{ default_iterator with
+
+  expr = (fun sub ({ exp_desc; exp_env; _ } as e) ->
+      (match exp_desc with
+      | Texp_ident (path, lid, _) ->
+          f ~namespace:Value exp_env path lid
+      | Texp_construct (lid, constr_desc, _) ->
+          add_constructor_description exp_env lid constr_desc
+      | Texp_field (_, lid, label_desc)
+      | Texp_setfield (_, lid, label_desc, _) ->
+          add_label exp_env lid label_desc
+      | Texp_new (path, lid, _) ->
+          f ~namespace:Class exp_env path lid
+      | Texp_record { fields; _ } ->
+        Array.iter (fun (label_descr, record_label_definition) ->
+          match record_label_definition with
+          | Overridden (
+              { Location.txt; loc},
+              {exp_loc; _})
+              when not exp_loc.loc_ghost
+                && loc.loc_start = exp_loc.loc_start
+                && loc.loc_end = exp_loc.loc_end ->
+            (* In the presence of punning we want to index the label
+                even if it is ghosted *)
+            let lid = { Location.txt; loc = {loc with loc_ghost = false} } in
+            add_label exp_env lid label_descr
+          | Overridden (lid, _) -> add_label exp_env lid label_descr
+          | Kept _ -> ()) fields
+      | Texp_instvar  (_self_path, path, name) ->
+          let lid = { name with txt = Longident.Lident name.txt } in
+          f ~namespace:Value exp_env path lid
+      | Texp_setinstvar  (_self_path, path, name, _) ->
+          let lid = { name with txt = Longident.Lident name.txt } in
+          f ~namespace:Value exp_env path lid
+      | Texp_override (_self_path, modifs) ->
+          List.iter (fun (id, (name : string Location.loc), _exp) ->
+            let lid = { name with txt = Longident.Lident name.txt } in
+            f ~namespace:Value exp_env (Path.Pident id) lid)
+            modifs
+
+      | Texp_constant _ | Texp_let _ | Texp_function _ | Texp_apply _
+      | Texp_match _ | Texp_try _ | Texp_tuple _ | Texp_variant _ | Texp_array _
+      | Texp_ifthenelse _ | Texp_sequence _ | Texp_while _ | Texp_for _
+      | Texp_send _
+      | Texp_letmodule _ | Texp_letexception _ | Texp_assert _ | Texp_lazy _
+      | Texp_object _ | Texp_pack _ | Texp_letop _ | Texp_unreachable
+      | Texp_extension_constructor _ | Texp_open _ -> ());
```

Comment:
There is small missing occurrence here: `Texp_extension_constructor` represents `[%extension_constructor Not_found]` and thus may hide an occurrence of an extension constructor.

Reply:
Good catch, I just pushed a test showing the missing occurrence and a fix.